### ISSUE TITLE
Field-values of 'null' for deeply nested models might become "out of sync"

### ISSUE THREAD

**Issue by toadle** [davidkpiano/react-redux-form#803]
### The Problem
I'm still trying to track down a state that I have in my application, where a validation on a deeply nested field causes a loop of `setFieldsValidity`-actions, up until `Maximum call stack size exceeded`. I know it is because of the validator, cause I can make it work when removing the validators.

While reconstructing my situation in a Pen I noticed this strange behaviour: When switching the `model` of a `<Form>`, e.g. for switching from a creating to an editing form: When the old model contained a value and the new model does not contain the value, the old value stays in the form. As if the "new" model's null value got ignored. Somehow the validators still tackle correctly, they display the error-message.

### Steps to Reproduce
See Codepen and push the "Toggle model"-button.

### Expected Behavior
The null-value should be reflected in the text-field.

### Actual Behavior
null value is not reflected and old 'present' value is kept.

### Reproducible Code Example
https://codepen.io/toadle/pen/aWrMxx

**Comment by davidkpiano** [comment]
The quick fix for this is changing this:

```diff
-lang: null,
+lang: '', // or lang: undefined
```

In fact, React will warn you if you try to use `null` for an input value. Changing it to an empty string or `undefined` seems to fix the behavior.

Please let me know if this also fixes your larger issue that you mentioned.

**Comment by toadle** [comment]
I fixed my loop, by adding this: 

```
  shouldComponentUpdate(nextProps) {
    return !_.isEqual(this.props, nextProps);
  }
```

...on the Component that contains the ´<Form>´.

So I suppose this somehow has to do with the `setFieldsValidity`-action that RRF dispatches, when the ´<Form>´ appears. But I can't tell if this is my stupidity or that needs fixing. 

I've tried to reconstruct my use-case: https://codepen.io/toadle/pen/aWrMxx

But in the above example it also doesn't loop. I access the model's value via ´connect´ and then depend the validators on it.
